espflash/src/error.rs

```diff
@@ -266,11 +266,9 @@ pub enum Error {
     )]
     PartitionTableDoesNotFit(FlashSize),
 
-    #[error(
-        "The app descriptor is not present in the project. You need to add the https://github.com/esp-rs/esp-hal/tree/main/esp-bootloader-esp-idf to your project."
-    )]
+    #[error("{error}")]
     #[diagnostic(code(espflash::app_desc::app_descriptor_not_present))]
-    AppDescriptorNotPresent,
+    AppDescriptorNotPresent { error: String },
```

Comment:
Sorry to nitpick, but there's no benefit to naming the inner value here

```suggestion
    AppDescriptorNotPresent(String),
```

The attribute then turns into 

```rust
#[error("{0}")]
```

And you no longer need to specify `error:` when constructing this error